pkg/operation/botanist/controlplane/kubescheduler/kube_scheduler_test.go

```diff
@@ -0,0 +1,472 @@
+				gomock.InOrder(
+					c.EXPECT().Get(ctx, kutil.Key(namespace, configMapName), gomock.AssignableToTypeOf(&corev1.ConfigMap{})),
+					c.EXPECT().Update(ctx, gomock.AssignableToTypeOf(&corev1.ConfigMap{})),
+					c.EXPECT().Get(ctx, kutil.Key(namespace, vpaName), gomock.AssignableToTypeOf(&autoscalingv1beta2.VerticalPodAutoscaler{})),
+					c.EXPECT().Update(ctx, gomock.AssignableToTypeOf(&autoscalingv1beta2.VerticalPodAutoscaler{})).Return(fakeErr),
+				)
+
+				Expect(kubeScheduler.Deploy(ctx)).To(MatchError(fakeErr))
+			})
+
+			It("should fail because the service cannot be created", func() {
+				gomock.InOrder(
+					c.EXPECT().Get(ctx, kutil.Key(namespace, configMapName), gomock.AssignableToTypeOf(&corev1.ConfigMap{})),
+					c.EXPECT().Update(ctx, gomock.AssignableToTypeOf(&corev1.ConfigMap{})),
+					c.EXPECT().Get(ctx, kutil.Key(namespace, vpaName), gomock.AssignableToTypeOf(&autoscalingv1beta2.VerticalPodAutoscaler{})),
+					c.EXPECT().Update(ctx, gomock.AssignableToTypeOf(&autoscalingv1beta2.VerticalPodAutoscaler{})),
+					c.EXPECT().Get(ctx, kutil.Key(namespace, serviceName), gomock.AssignableToTypeOf(&corev1.Service{})),
+					c.EXPECT().Update(ctx, gomock.AssignableToTypeOf(&corev1.Service{})).Return(fakeErr),
+				)
+
+				Expect(kubeScheduler.Deploy(ctx)).To(MatchError(fakeErr))
+			})
+
+			It("should fail because the deployment cannot be created", func() {
+				gomock.InOrder(
+					c.EXPECT().Get(ctx, kutil.Key(namespace, configMapName), gomock.AssignableToTypeOf(&corev1.ConfigMap{})),
+					c.EXPECT().Update(ctx, gomock.AssignableToTypeOf(&corev1.ConfigMap{})),
+					c.EXPECT().Get(ctx, kutil.Key(namespace, vpaName), gomock.AssignableToTypeOf(&autoscalingv1beta2.VerticalPodAutoscaler{})),
+					c.EXPECT().Update(ctx, gomock.AssignableToTypeOf(&autoscalingv1beta2.VerticalPodAutoscaler{})),
+					c.EXPECT().Get(ctx, kutil.Key(namespace, serviceName), gomock.AssignableToTypeOf(&corev1.Service{})),
+					c.EXPECT().Update(ctx, gomock.AssignableToTypeOf(&corev1.Service{})),
+					c.EXPECT().Get(ctx, kutil.Key(namespace, deploymentName), gomock.AssignableToTypeOf(&appsv1.Deployment{})),
+					c.EXPECT().Update(ctx, gomock.AssignableToTypeOf(&appsv1.Deployment{})).Return(fakeErr),
+				)
+
+				Expect(kubeScheduler.Deploy(ctx)).To(MatchError(fakeErr))
+			})
+
+			DescribeTable("success tests for various kubernetes versions",
+				func(version string, config *gardencorev1beta1.KubeSchedulerConfig) {
+					kubeScheduler, err = New(c, namespace, version, image, replicas, config)
+					Expect(err).NotTo(HaveOccurred())
+
+					kubeScheduler.SetSecrets(Secrets{
+						Kubeconfig: Secret{Name: secretNameKubeconfig, Checksum: secretChecksumKubeconfig},
+						Server:     Secret{Name: secretNameServer, Checksum: secretChecksumServer},
+					})
+
+					gomock.InOrder(
+						c.EXPECT().Get(ctx, kutil.Key(namespace, configMapName), gomock.AssignableToTypeOf(&corev1.ConfigMap{})),
+						c.EXPECT().Update(ctx, configMapFor(version)),
```

Comment:
Can you use the `DeepDerivativeEqual` to get the enhanced YAML diff?
```suggestion
						c.EXPECT().Update(ctx, gomock.AssignableToTypeOf(&corev1.ConfigMap{})).
							Do(func(ctx context.Context, obj runtime.Object, opts ...client.UpdateOption) {
								Expect(obj).To(DeepDerivativeEqual(configMapFor(version)))
							}),
```